component/config.go

```diff
@@ -29,15 +29,7 @@ import (
 // (e.g. check if a required field is present).
 //
 // A valid implementation MUST pass the check componenttest.CheckConfigStruct (return nil error).
-type Config interface {
-	// Deprecated: [v0.67.0] use Settings.ID.
-	ID() ID
-
-	// Deprecated: [v0.67.0] use Settings.ID.
-	SetIDName(idName string)
-
-	privateConfig()
-}
+type Config interface{}
```

Comment:
> // Config defines the configuration for a component.Component.

This description is probably not correct anymore, it can be applied to any configuration part now, right?

Do you have any further plans for this type? Do we even need it if it's just `interface{}` now?

Also `UnmarshalConfig` doesn't seems to be applicable to `compoonent` module anymore. Do you think we should keep it here?

Reply:
> This description is probably not correct anymore, it can be applied to any configuration part now, not component.Component only, right?

Not sure I understand. The "component.Factory.CreateDefaultConfig" returns this object. If others are using it somewhere else that is their problem... What are you looking for here?

> Do you have any further plans for this type? Do we even need it if it's just interface{} now?

Good discussion if we need it. We can think in a future PR about that. 

> Also UnmarshalConfig doesn't seems to be applicable to compoonent module anymore. Do you think we should keep it here?

Same as previous, we can followup with more changes, for the moment the purpose of https://github.com/open-telemetry/opentelemetry-collector/issues/4714 was to make sure we don't mutate the `Config` and that becomes an opaque object from the core collector perspective.

Reply:
> Not sure I understand. The "component.Factory.CreateDefaultConfig" returns this object. If others are using it somewhere else that is their problem... What are you looking for here?

Makes sense, so we have the Config still mostly applied to Component even if it can be used with any other sub-configuration. That's what I was looking for